pilot/pkg/config/kube/crd/conversion.go

```diff
@@ -78,7 +78,15 @@ func ConvertObject(schema collection.Schema, object IstioObject, domain string)
 	if err != nil {
 		return nil, err
 	}
-	data, err := FromJSON(schema, string(js))
+	spec, err := FromJSON(schema, string(js))
+	if err != nil {
+		return nil, err
+	}
+	js, err = json.Marshal(object.GetStatus())
+	if err != nil {
+		return nil, err
+	}
+	status, err := FromJSON(schema, string(js))
```

Comment:
Can we make sure status ends up as `nil` not an empty (but initialized) map when status is not set? To avoid extra allocations

Reply:
i think since we know its an `IstioCondition` we are marshalling into we dont need that function altogether